podpac/core/interpolation/interpolation_manager.py

```diff
@@ -0,0 +1,547 @@
+from __future__ import division, unicode_literals, print_function, absolute_import
+from copy import deepcopy
+from collections import OrderedDict
+from six import string_types
+import logging
+
+import traitlets as tl
+import numpy as np
+
+from podpac.core.units import UnitsDataArray
+from podpac.core.coordinates import merge_dims
+from podpac.core.interpolation.interpolator import Interpolator
+from podpac.core.interpolation.interpolators import NearestNeighbor, NearestPreview, Rasterio, ScipyPoint, ScipyGrid
+
+_logger = logging.getLogger(__name__)
+
+
+INTERPOLATION_DEFAULT = "nearest"
+"""str : Default interpolation method used when creating a new :class:`Interpolation` class """
+
+INTERPOLATORS = [NearestNeighbor, NearestPreview, Rasterio, ScipyPoint, ScipyGrid]
+"""list : list of available interpolator classes"""
+
+INTERPOLATORS_DICT = {}
+"""dict : Dictionary of a string interpolator name and associated interpolator class"""
+
+INTERPOLATION_METHODS = [
+    "nearest_preview",
+    "nearest",
+    "bilinear",
+    "cubic",
+    "cubic_spline",
+    "lanczos",
+    "average",
+    "mode",
+    "gauss",
+    "max",
+    "min",
+    "med",
+    "q1",
+    "q3",
+    "spline_2",
+    "spline_3",
+    "spline_4",
+]
+
+INTERPOLATION_METHODS_DICT = {}
+"""dict: Dictionary of string interpolation methods and associated interpolator classes
+   (i.e. ``'nearest': [NearestNeighbor, Rasterio, Scipy]``) """
+
+
+def load_interpolators():
+    """Load interpolators from :list:`INTERPOLATORS`
+
+    Defines :dict:`INTERPOLATORS_DICT`, and :dict:`INTERPOLATION_METHODS_DICT`
+    """
+
+    # create empty arrays in INTEPROLATOR_METHODS
+    for method in INTERPOLATION_METHODS:
+        INTERPOLATION_METHODS_DICT[method] = []
+
+    # fill dictionaries with interpolator properties
+    for interpolator_class in INTERPOLATORS:
+        interpolator = interpolator_class()
+        INTERPOLATORS_DICT[interpolator.name] = interpolator_class
+
+        for method in INTERPOLATION_METHODS:
+            if method in interpolator.methods_supported:
+                INTERPOLATION_METHODS_DICT[method].append(interpolator_class)
+
+
+# load interpolators when module is first loaded
+# TODO does this really only load once?
+# TODO maybe move this whole section?
+load_interpolators()
+
+
+class InterpolationException(Exception):
+    """
+    Custom label for interpolation exceptions
+    """
+
+    pass
+
+
+class InterpolationManager(object):
```

Comment:
I agree... BUT for this PR I didn't want to touch it too much so that I could avoid digging too deeply into the guts. I plan to do that on stage 3 -- making the selectors more useful. Ok to wait till then?